tools/rna_tools/vienna_rna/rnafold.xml

```diff
@@ -17,33 +22,47 @@
     #else
         $measelect.pf
     #end if
-    #if $varExists('$advancedOptions.nogu'):
-        $advancedOptions.noconversion
-        $advancedOptions.gquad
-        $advancedOptions.nolp
-        $advancedOptions.nogu
-        $advancedOptions.noclosinggu
-        $advancedOptions.canonicalonly
-        $advancedOptions.circular
+    
+    $advancedOptions.noconversion
+    $advancedOptions.gquad
+    $advancedOptions.nolp
+    $advancedOptions.nogu
+    $advancedOptions.noclosinggu
+    $advancedOptions.canonicalonly
+    $advancedOptions.circular
+
+    < 
+    #if str($input_source.select_fasta) == "false"
+        "input.fasta"
+    #else
+        "${input_source.fasta_input}"
     #end if
 
-    < $fasta_input
-
     > $tabular_file
 ]]>
     </command>
 
     <inputs>
-    <param format="fasta" name="fasta_input" type="data" label="FASTA file"/>
-        <param name="temperature" type="float" value="37.0" label="Temperature [°C]" help="-T"/>
-        <param name="dangling" type="select" label="how to treat dangling end energies" help="-d">
+        <conditional name="input_source">
+            <param name="select_fasta" type="boolean" truevalue="true" falsevalue="false" label="Input from FASTA file" selected="false" />
+
+            <when value="true">
+                <param format="fasta" name="fasta_input" type="data" label="Sequence to fold (FASTA file)"/>
+            </when>
+            <when value="false">
+                <param name="input_sequence" type="text" label="Input sequence"/>
```

Comment:
Hopefully no one will enter a fasta sequence here. Can we be smarter at the cheetah level and look at the first character? Something like $input.strip()[0] != '>'?